pkg/operator/controller/ingress/controller.go

```diff
@@ -907,10 +952,23 @@ func (r *reconciler) ensureIngressController(ci *operatorv1.IngressController, d
 		errs = append(errs, fmt.Errorf("failed to list pods in namespace %q: %v", operatorcontroller.DefaultOperatorNamespace, err))
 	}
 
-	errs = append(errs, r.syncIngressControllerStatus(ci, deployment, deploymentRef, pods.Items, lbService, operandEvents.Items, wildcardRecord, dnsConfig, infraConfig))
+	if syncStatusErr, updated := r.syncIngressControllerStatus(ci, deployment, deploymentRef, pods.Items, lbService, operandEvents.Items, wildcardRecord, dnsConfig, infraConfig); syncStatusErr != nil {
+		errs = append(errs, syncStatusErr)
+		// If syncIngressControllerStatus updated our ingress status, it's important we query for that new object.
+		// If we don't, then the next function syncRouteStatus would always fail because it has a stale ingress object.
+		if updated {
+			updatedIc := &operatorv1.IngressController{}
+			if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: ci.Namespace, Name: ci.Name}, updatedIc); err != nil {
+				errs = append(errs, fmt.Errorf("failed to get ingresscontroller: %w", err))
+			}
+			ci = updatedIc
+		}
+	}
```

Comment:
Why do you only update `ci` if `syncStatusErr != nil`?  Couldn't `syncIngressControllerStatus` update the status, return `nil, true`, and cause `syncRouteStatus` to fail its update?

Reply:
BTW, `NewMaybeRetryableAggregate` ignores nil values.  I think you want the following:
```suggestion
	syncStatusErr, updated := r.syncIngressControllerStatus(ci, deployment, deploymentRef, pods.Items, lbService, operandEvents.Items, wildcardRecord, dnsConfig, infraConfig)
	errs = append(errs, syncStatusErr)
	// If syncIngressControllerStatus updated our ingress status, it's important we query for that new object.
	// If we don't, then the next function syncRouteStatus would always fail because it has a stale ingress object.
	if updated {
		updatedIc := &operatorv1.IngressController{}
		if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: ci.Namespace, Name: ci.Name}, updatedIc); err != nil {
			errs = append(errs, fmt.Errorf("failed to get ingresscontroller: %w", err))
		}
		ci = updatedIc
	}
```

Reply:
Yep, I removed this logic and double checked I was doing something valid. It works without with this logic removed, but I get this error on the first reconcilation :
>  failed to update ingresscontroller status: Operation cannot be fulfilled on ingresscontrollers.operator.openshift.io \"ic-delete-test\": the object has been modified; please apply your changes to the latest version and try again

So just a sanity check for why I did this.